src/hooks/useTopVoters.ts

```diff
@@ -0,0 +1,41 @@
+import useAsyncMemo from 'decentraland-gatsby/dist/hooks/useAsyncMemo'
+
+import { SNAPSHOT_SPACE } from '../entities/Snapshot/constants'
+
+import { Snapshot } from './../api/Snapshot'
+
+import { useSortingByKey } from './useSortingByKey'
+
+type Voters = {
+  address: string
+  votes: number
+}
+
+export default function useTopVoters(start: Date, end: Date, limit: number) {
+  const [votes, state] = useAsyncMemo(
+    async () => {
+      const votes = await Snapshot.get().getVotes(SNAPSHOT_SPACE, start, end)
+      const votesByUser = votes.reduce((acc, vote) => {
+        const address = vote.voter.toLowerCase()
+        acc[address] = (acc[address] || 0) + 1
+        return acc
+      }, {} as Record<string, number>)
+
+      return Object.entries(votesByUser).map<Voters>(([address, votes]) => ({ address, votes }))
+    },
+    [limit],
```

Comment:
I don't think the asyncMemo should re-evaluate depending on the limit change, because you are not using it to fetch different data (it only depends on the start and end date). And limit is only used in the slice at the end of the useTopVoters hook.